pkg/oc/admin/prune/imageprune/prune.go

```diff
@@ -398,10 +422,18 @@ func (p *pruner) addImageStreamsToGraph(streams *imageapi.ImageStreamList, limit
 					}
 
 					glog.V(4).Infof("Adding reference from stream %s to %s", getName(stream), cn.Describe())
-					if cn.Type == imagegraph.ImageComponentTypeConfig {
+					switch cn.Type {
+					case imagegraph.ImageComponentTypeConfig:
 						p.g.AddEdge(imageStreamNode, s, ReferencedImageConfigEdgeKind)
-					} else {
+						break
+					case imagegraph.ImageComponentTypeLayer:
 						p.g.AddEdge(imageStreamNode, s, ReferencedImageLayerEdgeKind)
+						break
+					case imagegraph.ImageComponentTypeManifest:
+						p.g.AddEdge(imageStreamNode, s, ReferencedImageManifestEdgeKind)
+						break
+					default:
+						panic(fmt.Sprintf("unhandeled image component type %q", cn.Type))
```

Comment:
Can we return this error?

Reply:
Well, I don't expect this to ever fire. But sure.

Reply:
also should be "unhandled"

but yeah, i don't think we ever want to panic, even in the cli.

Reply:
Fixed.